### PR TITLE
composeBox, iOS: Fix msg input is not scrolled on typing.

### PR DESCRIPTION
**Pull request by jainkuniya**
Now this is how it looks:

Fixes: #3614
This reverts #3595 and include fix for #3369.

image
image

### CODE REVIEW

**Comment by gnprice** [comment]
Thanks @jainkuniya !

From the commit messages, I think this is a WIP branch. (There's a "TODO add more details here" in one of them :wink:) It's best to make that explicit in the PR thread too.

There's an issue this fixes, right? Please mention it in the PR description, and in the commit message.

---

It looks like what this does in the code is:
* add a wrapper `View` around each of the topic input and message-content input
* add `justifyContent: 'space-around'` to the style for the common wrapper `View` we have for those two inputs together
* replace `marginBottom` on the topic input with `marginTop` on the new wrapper `View` of the message-content input

It's not obvious to me why any of those is related to the issue. Do you have a theory for why this changes the behavior?

Also, are all three (or more) of the changes actually relevant? Try removing bits of it one at a time:
* no new wrapper `View` for the topic input
* keep the `marginBottom` rather than `marginTop`
* no new wrapper `View` for the content input -- can put the style on the input itself instead of wrapper
* etc.

If a simpler change works just as well, then it may help for understanding what actually drives it -- which will make it easier to avoid introducing a similar issue in the future. Also makes a smaller, easier-to-understand diff :smile:

**Comment by jainkuniya** [comment]
Ya, will add more details next.

Actually this current state is just based on try and run 😅. Not sure about the exact reasons, will try to find them out.

Just curious, have you tried this branch?

**Comment by gnprice** [comment]
I haven't yet tried the branch; it seems like this is still in the investigation phase, and I'm relying on you to do the investigation :slightly_smiling_face: 

I'll definitely be interested in hearing what results you get when you try removing bits of this change one at a time (including the bits I suggested above). Among other reasons, it's likely that will help us understand what's going on.

**Comment by gnprice** [comment]
From our chat discussion, I now understand your screenshots better: those are both from a stream narrow. In the second one, the message content input has grown so tall it's covering over the topic input.

Trying the branch out, I got a slightly different result!
image
That's on a simulated iPhone 8 running iOS 13.2.2. The content input has grown tall... but it's partly *under* the topic input, instead of over it. (It seems to be over some of the padding of the topic input, but under the topic input itself.)

So that's interesting that it behaves differently. Can you describe what conditions you're using that get the result you showed above?

Both of these behaviors seem like they're improvements on the status quo, so I'd be OK merging a version that still has this behavior.

I would still like you to do the experiment of removing bits of this change one at a time. That will be informative, and may make the change simpler.

File: src/compose/ComposeBox.js [hunk 1]
```diff
@@ -292,7 +292,9 @@ class ComposeBox extends PureComponent<Props, State> {
       alignItems: 'flex-end',
     },
     composeText: {
+      flex: 1,
       paddingVertical: 8,
+      justifyContent: 'flex-end',
```
Comment (jainkuniya) [inline src/compose/ComposeBox.js]: Have never imagined fix will be this short 🤦‍♂ 

Now I guess this solution is ideal as we wanted 😍

**Comment by gnprice** [comment]
Summarizing our conversation just now:

It looks like the core issue here is that the content input is too tall. In the extreme case the input is too tall for the whole input to be shown at all; in other cases it's itself shown in full, but it crowds out the topic input, or the unreads banner.

The iOS screenshots above are on an iPhone X-family model, with a notch, which pulls in an unrelated bug that causes overlapping symptoms. Best to focus on a model without a notch. I use a simulated iPhone 8. On that, this branch still shows symptoms caused by the input being too tall.

It seems like we still don't yet really understand what's causing the issue, or why these attempted fixes have the effects they do. I'd like you to focus at this stage on investigating the issue so we can understand what's going on, and not on the code of an attempted fix. That's best tracked over on the issue thread #3614; I'll close this PR.

**Pull request closed without merging**